sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -479,6 +479,7 @@ private[joins] object UnsafeHashedRelation {
           throw QueryExecutionErrors.cannotAcquireMemoryToBuildUnsafeHashedRelationError()
         }
       } else if (isNullAware) {
+        binaryMap.free()
         return HashedRelationWithAllNullKeys
```

Comment:
shall we return it before creating the binary map?

Reply:
@cloud-fan - the NULL-aware ANTI join will be done if there's row with NULL join keys. Note the `if (!key.anyNull || allowsNullKey)` at line 472. So it will be taking effect when we reading the row with NULL join keys, and this information is not known before creating the binary map